src/record/record.js

```diff
@@ -427,9 +419,9 @@ Record.prototype._onRead = function( message ) {
  */
 Record.prototype._setReady = function() {
 	this.isReady = true;
-	for( var i = 0; i < this._queuedMethodCalls.length; i++ ) {
-		this[ this._queuedMethodCalls[ i ].method ].apply( this, this._queuedMethodCalls[ i ].args );
-	}
+	this._queuedMethodCalls.forEach( function(call) {
```

Comment:
We don't use for each functions on hot paths. They unfortunately are not optimised as well as for loops, both from a memory and CPU aspect.